src/Symfony/Component/Ldap/Adapter/ExtLdap/EntryManager.php

```diff
@@ -159,7 +159,7 @@ public function applyOperations(string $dn, iterable $operations): void
 
     private function parseRdnFromEntry(Entry $entry): string
     {
-        if (!preg_match('/^([^,]+),/', $entry->getDn(), $matches)) {
+        if (!preg_match('/(^[^,\\\\]*(?:\\\\.[^,\\\\]*)*),/', $entry->getDn(), $matches)) {
```

Comment:
the regexp won't work when several `\` are used, eg with `\\\\` or `\\\\\,`,isn't it?

Reply:
The number of `\` shouldn't matter. Compared to the old code, basically the same is done, except that now the comma is not matched if it is escaped with a `\`. 

It's a bit tricky with the "not matching regex groups" (starting with ?: ). It is quite hard to read and understand.